src/mbgl/style/expression/match.cpp

```diff
@@ -40,6 +40,20 @@ std::vector<optional<Value>> Match<T>::possibleOutputs() const {
     return result;
 }
 
+template <typename T>
+mbgl::Value Match<T>::serialize() const {
+    static const std::string match = "match";
+    std::vector<mbgl::Value> serialized;
+    serialized.emplace_back(match);
+    serialized.emplace_back(input->serialize());
+    for (auto& entry : branches) {
+        serialized.emplace_back(entry.first);
+        serialized.emplace_back(entry.second->serialize());
+    };
+    serialized.emplace_back(otherwise->serialize());
+    return serialized;
+}
```

Comment:
Note that this could make serialize some expressions much more verbosely:
```js
[
  "match",
  [1, 2, 3, 4, 5], some_complex_expression,
  "otherwise"
]
```
would become:
```js
[
  "match",
  1, some_complex_expression,
  2, some_complex_expression,
  3, some_complex_expression,
  4, some_complex_expression,
  5, some_complex_expression,
  "otherwise"
]
```

That said, I'm not sure whether it's really worth it to produce the more compact form.

Reply:
Yeah, I bucketed this under "OK as long as it's semantically equivalent". I'm inclined to leave as is in the name of simplicity, unless you have a hunch that we may actually see a fair number of expressions that blow up in size with the flattened format...

Reply:
> unless you have a hunch that we may actually see a fair number of expressions that blow up in size with the flattened format 

I wasn't thinking we needed to worry about this, but on second thought, I think we actually probably should consolidate 😬